test/SourceKit/Swiftify/clang-overload-cursor-info.swift

```diff
@@ -0,0 +1,41 @@
+// REQUIRES: swift_feature_SafeInteropWrappers
+// REQUIRES: swift_feature_LifetimeDependence
+
+import FromClang
+
+// RUN: %target-swift-ide-test \
+// RUN:   -print-module -module-to-print=FromClang -source-filename=x \
+// RUN:   -plugin-path %swift-plugin-dir -I %S/Inputs \
+// RUN:   -enable-experimental-feature SafeInteropWrappers \
+// RUN:   -enable-experimental-feature LifetimeDependence \
+// RUN: | %FileCheck %s --check-prefix INTERFACE
+
+// The macro-generated interface we're looking up source info for
+// (this is more so for documentation than checking correctness)
+//
+// INTERFACE:      @_alwaysEmitIntoClient @_disfavoredOverload public func hasBufferOverload(_ p: UnsafeMutableBufferPointer<Int32>)
+// INTERFACE:      @lifetime(p: copy p)
+// INTERFACE-NEXT: @_alwaysEmitIntoClient @_disfavoredOverload public func hasSpanOverload(_ p: inout MutableSpan<Int32>)
+
+// RUN: %sourcekitd-test -req=cursor -pos=26:3 %s -- -I %S/Inputs %s \
+// RUN:   -enable-experimental-feature SafeInteropWrappers \
+// RUN:   -enable-experimental-feature LifetimeDependence \
+// RUN: | %FileCheck %s --check-prefix CLANG-HEADER
+@inlinable
+public func callWithBufferPtr(_ p: UnsafeMutableBufferPointer<CInt>) {
+  hasBufferOverload(p)
+}
+
+// RUN: %sourcekitd-test -req=cursor -pos=37:3 %s -- -I %S/Inputs %s \
+// RUN:   -enable-experimental-feature SafeInteropWrappers \
+// RUN:   -enable-experimental-feature LifetimeDependence \
+// RUN: | %FileCheck %s --check-prefix CLANG-HEADER
+@available(visionOS 1.1, tvOS 12.2, watchOS 5.2, iOS 12.2, macOS 10.14.4, *)
+@lifetime(p: copy p)
+@inlinable
+public func callReturnLifetimeBound(_ p: inout MutableSpan<CInt>) {
+  hasSpanOverload(p)
+}
+
+// Check that we get the file path of the header that Clang imported:
+// CLANG-HEADER: /Inputs/from-clang.h:
```

Comment:
Does the output contain a source range also? It'd be good to verify the location. You can use e.g. `[[@LINE+1]]` in FileCheck to make it less sensitive to source changes

Reply:
That's a good idea. I'll need to put the check statements in the header itself but that's probably a good idea anyway.

Reply:
I'll just shove my vote in here for split-file. Makes tests sooo much easier to read :)